File: 0693_binary_number_with_alternating_bits/Solution.java
// LeetCode 0693 - Binary Number with Alternating Bits
// https://leetcode.com/problems/binary-number-with-alternating-bits/

class Solution {
    public boolean hasAlternatingBits(int n) {
        int x = n ^ (n >>> 1);
        return (x & (x + 1)) == 0;
    }
}

File: 0693_binary_number_with_alternating_bits/solution.rs
// LeetCode 0693 - Binary Number with Alternating Bits
// https://leetcode.com/problems/binary-number-with-alternating-bits/

impl Solution {
    pub fn has_alternating_bits(n: i32) -> bool {
        let x = (n as u32) ^ ((n as u32) >> 1);
        x & (x + 1) == 0
    }
}
